Library/Formula/nghttp2.rb

```diff
@@ -87,35 +89,66 @@ class Nghttp2 < Formula
     sha256 "3e15b416c9a2039c1a51208b2cd3bb4ffd796cd19e601b1d2657afcb77c3dc90"
   end
 
+  resource "Cython" do
+    url "https://pypi.python.org/packages/source/C/Cython/cython-0.22.tar.gz"
+    sha256 "14307e7a69af9a0d0e0024d446af7e51cc0e3e4d0dfb10d36ba837e5e5844015"
+  end if build.with?("python3")
+
   # https://github.com/tatsuhiro-t/nghttp2/issues/125
   # Upstream requested the issue closed and for users to use gcc instead.
   # Given this will actually build with Clang with cxx11, just use that.
   needs :cxx11
 
+  def python_version
+   @python_version ||= begin
+     build.with?('python3') ? [3, 4] : [2, 7]
+   end
+  end
+
+  def python
+   @python ||= begin
+     "python#{python_version.join('.')}"
+   end
+  end
+
+  patch :p1, :DATA
```

Comment:
Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!